p2p/host/autorelay/relay_finder.go

```diff
@@ -119,6 +131,8 @@ func (rf *relayFinder) background(ctx context.Context) {
 	}
 	defer subConnectedness.Close()
 
+	peerSourceTimer := rf.conf.clock.Timer(rf.conf.minInterval)
```

Comment:
Would it make sense to have a single timer that gets set to `min(nextRefresh, nextBackoff, ...)`? You could then deal with resetting this timer exclusively in this function.

Reply:
We can try and see if that's simpler to understand.

Reply:
I think it's better. It has the nice side effect of avoiding the `/ 5` timers. We also have a single timer now. And a single channel to rate limit findNodes.